src/rattler_build_conda_compat/jinja/jinja.py

```diff
@@ -4,42 +4,78 @@
 
 import jinja2
 import yaml
-from jinja2 import DebugUndefined
 
+from rattler_build_conda_compat.jinja.filters import _bool, _split, _version_to_build_string
+from rattler_build_conda_compat.jinja.objects import (
+    _stub_compatible_pin,
+    _stub_is_linux,
+    _stub_is_unix,
+    _stub_is_win,
+    _stub_match,
+    _stub_subpackage_pin,
+    _StubEnv,
+)
+from rattler_build_conda_compat.jinja.utils import _MissingUndefined
 from rattler_build_conda_compat.loader import load_yaml
 
 
 class RecipeWithContext(TypedDict, total=False):
     context: dict[str, str]
 
 
-class _MissingUndefined(DebugUndefined):
-    def __str__(self) -> str:
-        """
-        By default, `DebugUndefined` return values in the form `{{ value }}`.
-        `rattler-build` has a different syntax, so we need to override this method,
-        and return the value in the form `${{ value }}`.
-        """
-        return f"${super().__str__()}"
-
-
 def jinja_env() -> jinja2.Environment:
     """
     Create a `rattler-build` specific Jinja2 environment with modified syntax.
+    Target platform, build platform, and mpi are set to linux-64 by default.
     """
-    return jinja2.Environment(
+
+    env = jinja2.Environment(
         variable_start_string="${{",
         variable_end_string="}}",
         trim_blocks=True,
         lstrip_blocks=True,
-        autoescape=True,
+        autoescape=jinja2.select_autoescape(default_for_string=False),
         undefined=_MissingUndefined,
     )
 
+    env_obj = _StubEnv()
+
+    # inject rattler-build recipe functions in jinja environment
+    env.globals.update(
+        {
+            "compiler": lambda x: x + "_compiler_stub",
+            "stdlib": lambda x: x + "_stdlib_stub",
+            "pin_subpackage": _stub_subpackage_pin,
+            "pin_compatible": _stub_compatible_pin,
+            "cdt": lambda *args, **kwargs: "cdt_stub",  # noqa: ARG005
+            "env": env_obj,
+            "match": _stub_match,
+            "is_unix": _stub_is_unix,
+            "is_win": _stub_is_win,
+            "is_linux": _stub_is_linux,
+            "unix": True,
+            "linux": True,
+            "target_platform": "linux-64",
+            "build_platform": "linux-64",
+            "mpi": "mpi",
+        }
+    )
+
+    # inject rattler-build recipe filters in jinja environment
+    env.filters.update(
+        {
+            "version_to_buildstring": _version_to_build_string,
+            "split": _split,
```

Comment:
I actually don't know if this is implemented in Python Jinja or not. Is it missing from Python jinja?

Reply:
It was missing from python jinja